casepro/msgs/tasks.py

```diff
@@ -91,3 +103,119 @@ def reply_export(export_id):
     logger.info("Starting replies export #%d..." % export_id)
 
     ReplyExport.objects.get(pk=export_id).do_export()
+
+
+def create_faq(org, question, answer, language, parent, labels=(), **kwargs):
+    """
+    A helper for creating FAQs since labels (many-to-many) needs to be added after initial creation
+    """
+    faq = FAQ.objects.create(org=org, question=question, answer=answer, language=language, parent=parent, **kwargs)
+    faq.labels.add(*labels)
+    return faq
+
+
+def get_or_create_lang(org, lang_code):
+    """
+    Gets or creates a Language object from a potentially poorly formatted language code
+    """
+    lang_code = lang_code.strip()
+    try:
+        # Get the Language
+        return Language.objects.get(code__iexact=lang_code)  # iexact removes case sensitivity
+    except:
+        # Format the lang_code before creating a new language
+        if len(lang_code) != 6:
+            raise ValueError("Language codes should be 6 characters long")
+
+        if lang_code[3] != '_':
+            raise ValueError("Language name and location should be seperated by an underscore")
+
+        lang_code = lang_code[0:4] + lang_code[4:6].upper()
+        # Create the language
+        return Language.objects.create(org=org, code=lang_code)
+
+
+def get_labels(labelstring):
+    """
+    Gets a list of label objects from a comma-seperated string of the label codes, eg. "TB, aids"
+    """
+    labels = set()
+    labelstrings = labelstring.split(', ')
+    print labelstrings
+    for labelstring in labelstrings:
+        labelstring = labelstring.strip()
+        print labelstring
+        try:
+            label = Label.objects.get(name__iexact=labelstring)  # iexact removes case sensitivity
+        except Exception as e:
+            task.log('Label does not exist! Create it first.')
+            raise e
+        print label
+        labels.add(label)
+    return list(labels)
+
+
+@task(track_started=True)
+def faq_csv_import(org, task_id):  # pragma: no cover
+    task = ImportTask.objects.get(pk=task_id)
+    log = StringIO()
+
+    task.task_id = faq_csv_import.request.id
+    task.log("Started import at %s" % timezone.now())
+    task.log("--------------------------------")
```

Comment:
You mean other than this: https://github.com/praekelt/casepro/pull/12/files#diff-96833113def594a421fdef9bac214d27R209 ?